### ISSUE TITLE
Prebid Generic Adapter 

### ISSUE THREAD

**Issue by johndpknt** [prebid/prebid-server#1148]
Hello Team,

I have been using the prebid generic adapter in prebid.js to configure and call our own hosted prebid-server over  ORTB 2.5 protocol, I would like to understand is there any generic adapter in prebid-server which can be used to make vanilla ORTB 2.5 calls to any other prebid-server? if no how the community will suggest me to go ahead with this thought process?

Thanks in advance

**Comment by SyntaxNode** [comment]
There is not currently a generic adapter in Prebid Server. Would you expect it to be drive solely from configuration changes or a simple code change? The code changes I have in mind are entries in

- https://github.com/prebid/prebid-server/blob/547b6ff619489a240e644f34fda4ee03e05f3b47/exchange/adapter_map.go#L64
- https://github.com/prebid/prebid-server/blob/547b6ff619489a240e644f34fda4ee03e05f3b47/usersync/usersyncers/syncer.go#L60

**Comment by bretg** [comment]
I'd be ok supporting a generic ortb adapter if we can solve the usersync feature -- probably in a config file. Auction requests could be supported with an alias

{code}
  "ext": {
    "prebid": {
      "aliases": {
        "myservice": "openrtb"
      }
    }
{code}
The requirements from my perspective are:
- it cannot be possible for the client to provide an arbitrary URL that Prebid Server invokes. It needs to be configurable on the server side so it can be tested and blocked as needed
- analytics needs to have access to the alias for reporting
- there should be separate metrics logged for the actual adapter and the aliases
- we should consider per-bidder controls (e.g. GDPR) to confirm whether they should be controlled at the entire generic adapter level or the alias level.

We'll discuss this feature in the next Prebid Server meeting.

**Comment by bretg** [comment]
@johndpknt - we discussed today in the Prebid Server meeting and would like to better understand your scenario. This statement isn't clear to us:

> can be used to make vanilla ORTB 2.5 calls to any other prebid-server

First, vanilla ORTB isn't really a thing -- there are always bidder-specific parameters of some kind. Second, are you talking about PBS calling another PBS?

We've discussed a generic adapter in the past as a way of making it easier for SSPs to plug into Prebid Server, but don't currently view that as a worthwhile effort. A generic adapter would need to parse JSON configuration defining the endpoint, usersync, and bidder parameter types. Because of that run-time parsing, it would be somewhat slower and burn more CPU than a native adapter.

Are you aware of demand sources that would in fact benefit from a lower technical barrier-to-entry?
